src/app/SubscriptionResumptionHelper.cpp

```diff
@@ -0,0 +1,80 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app/InteractionModelEngine.h>
+#include <app/SubscriptionResumptionHelper.h>
+
+namespace chip {
+namespace app {
+SubscriptionResumptionHelper::SubscriptionResumptionHelper() :
+    mOnConnectedCallback(HandleDeviceConnected, this), mOnConnectionFailureCallback(HandleDeviceConnectionFailure, this)
+{}
+
+CHIP_ERROR SubscriptionResumptionHelper::ResumeSubscription(CASESessionManager & caseSessionManager,
+                                                            SubscriptionResumptionStorage::SubscriptionInfo & subscriptionInfo)
+{
+    mNodeId         = subscriptionInfo.mNodeId;
+    mFabricIndex    = subscriptionInfo.mFabricIndex;
+    mSubscriptionId = subscriptionInfo.mSubscriptionId;
+    mMinInterval    = subscriptionInfo.mMinInterval;
+    mMaxInterval    = subscriptionInfo.mMaxInterval;
+    mFabricFiltered = subscriptionInfo.mFabricFiltered;
+    mAttributePaths = std::move(subscriptionInfo.mAttributePaths);
+    mEventPaths     = std::move(subscriptionInfo.mEventPaths);
```

Comment:
maybe a "copy from a subscriptioninfo into self" should be some "operator=" overload.

Do you need to derive from SubscriptionInfo? Why not use composition instead.

Reply:
I make the subscriptionInfo a member of this class. should I add the "operator=" overload in the SubscriptionInfo class?

Reply:
if copying from subscriptioninfo makes sense, an operator= in there seems reasonable.